src/Http/Middleware/AuthenticateSession.php

```diff
@@ -64,8 +64,10 @@ public function handle(Request $request, Closure $next): Response
         }
 
         return tap($next($request), function () use ($request, $guards) {
-            if (! is_null($request->user())) {
-                $this->storePasswordHashInSession($request, $guards->keys()->first());
+            $guard = $guards->keys()->first();
+
+            if ($this->auth->guard($guard)->hasUser()) {
```

Comment:
Sanctum can be configured to attempt to use an array of guards to authenticate users. How do you know the first guard in the list is the one that actually was used and is the one that has the user?

Reply:
good point :eyes:. i did not give it much thought since it was like that before. i rewrote it to getting first guard that user is authenticated with.